src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java

```diff
@@ -427,8 +427,15 @@ private void addInstanceCollectors(List<Collector> activeCollectors, List<Server
             String finalDirSuffix = Paths.get(dirSuffix, server.getName()).toString();
             String instanceType = instanceWithType.get(server.getName());
 
-            if (domainXml && instanceType.equals("CONFIG")) {
-                activeCollectors.add(new DomainXmlCollector(Paths.get(domainUtil.getNodePaths().get(server.getNodeRef()).toString(), server.getName(), "config", "domain.xml"), server.getName(), finalDirSuffix, obfuscateDomainXml, this));
+            if (domainXml) {
+                if (instanceType.equals("CONFIG")){
```

Comment:
You can improve this by using switch statement with Strings
```suggestion
                switch(instanceType) {
                    case "CONFIG":
                     activeCollectors.add(new DomainXmlCollector(Paths.get(domainUtil.getNodePaths().get(server.getNodeRef()).toString(), server.getName(), "config", "domain.xml"), server.getName(), finalDirSuffix, obfuscateDomainXml, this));
                    break;
                    case "SSH":
                      Path domainXmlPath = Paths.get((String) parameterMap.get(DOMAIN_XML_FILE_PATH));
                    activeCollectors.add(new DomainXmlCollector(domainXmlPath, server.getName(), finalDirSuffix, obfuscateDomainXml, this));
                    break;
                    default:
                      //somehting to do here
                }
```